basic_samples/IngressClientLibraries/DotNet/Program.cs

```diff
@@ -102,7 +113,37 @@ private static async Task MainAsync()
                 Console.WriteLine();
 
                 // At this point, we are ready to send OMF data to OCS.
-                // Please visit https://github.com/osisoft/OMF-Samples/tree/master/Tutorials/CSharp_Sds to learn how to do this.
+                //create type
+                Console.WriteLine($"Creating Type with Id {SimpleOMFType.TypeId}");
+                Console.WriteLine();
+                string OMFTypeBody = OMFGenerator.ProduceOMFTypeMessage(new List<JObject>() { OMFGenerator.GetTypeInOMF<SimpleOMFType>(new OMFTypeDefinition(SimpleOMFType.TypeId)) });
+                OMFMessage omfType = OMFHelper.GenerateOMFTypeMessage(OMFTypeBody, MessageAction.Create);
+                await deviceIngressService.SendOMFMessageAsync(omfType);
+
+                //create container
+                Console.WriteLine($"Creating Container with Id {streamId}");
+                Console.WriteLine();
+                string OMFContainerBody = OMFGenerator.ProduceOMFContainerMessage(new List<OMFContainerDefinition>() { new OMFContainerDefinition(streamId, SimpleOMFType.TypeId) });
+                OMFMessage omfContainer = OMFHelper.GenerateOMFContainerMessage(OMFContainerBody, MessageAction.Create);
+                await deviceIngressService.SendOMFMessageAsync(omfContainer);
+
+                //send random data points for 30 seconds
+                Random rand = new Random();
+                Console.WriteLine("Sending OMF Data Messages. Pres ESC key to stop sending.");
+                do
+                {
+                    while (!Console.KeyAvailable)
+                    {
+                        SimpleOMFType dataPoint = new SimpleOMFType() { Timestamp = DateTime.UtcNow, Value = rand.NextDouble() };
+                        string OMFDataBody = OMFGenerator.ProduceOMFDataMessage(new List<OMFValuesGroup>() { new OMFValuesPerContainer(streamId, new List<SimpleOMFType>() { dataPoint }) });
+                        OMFMessage omfDataMessage = OMFHelper.GenerateOMFDataMessage(OMFDataBody, MessageAction.Create);
+
+                        await deviceIngressService.SendOMFMessageAsync(omfDataMessage);
+                        Console.WriteLine($"Sent data point: Time: {dataPoint.Timestamp}, Value: {dataPoint.Value}");
+                        Task.Delay(1000).Wait();
```

Comment:
Might want to use -- await Task.Delay(1000)

Reply:
updated